ts/src/main/scala/ch/epfl/ts/component/Component.scala

```diff
@@ -36,15 +36,19 @@ final class ComponentBuilder(name: String) {
   } }
 
   def createRef(props: ComponentProps, name: String) = {
-    instances = new ComponentRef(system.actorOf(props, name), props.clazz, name) :: instances
+    instances = new ComponentRef(system.actorOf(props, name), props.clazz, name, this) :: instances
     instances.head
   }
 }
 
-class ComponentRef(val ar: ActorRef, val clazz: Class[_], val name: String) {
+class ComponentRef(val ar: ActorRef, val clazz: Class[_], val name: String, cb: ComponentBuilder) {
   // TODO: Verify clazz <: Component
-  def addDestination(destination: ComponentRef, data: Class[_])(implicit cb: ComponentBuilder) = {
-    cb.add(this, destination, data)
+  def ->(destination: ComponentRef, types: Class[_]*) = {
```

Comment:
Very expressive syntax :+1: 

Could you add JavaDoc to explain this function's behavior?